tests/std/tests/P2505R5_monadic_functions_for_std_expected/test.cpp

```diff
@@ -0,0 +1,332 @@
+        Expected with_value{in_place, 42};
+        const Expected const_with_value{in_place, 1337};
+        assert(with_value.error_or(payload_error_or{1}) == 1 + 3);
+        assert(const_with_value.error_or(input) == 2 + 2);
+        static_assert(noexcept(with_value.error_or(payload_error_or{1})) == construction_is_noexcept);
+        static_assert(noexcept(const_with_value.error_or(input)) == construction_is_noexcept);
+
+        assert(move(with_value).error_or(payload_error_or{1}) == 1 + 3);
+        assert(move(const_with_value).error_or(input) == 2 + 2);
+        static_assert(noexcept(move(with_value).error_or(payload_error_or{1})) == construction_is_noexcept);
+        static_assert(noexcept(move(const_with_value).error_or(input)) == construction_is_noexcept);
+    }
+
+    { // with convertible argument
+        using Expected = expected<int, payload_error_or>;
+
+        Expected with_error{unexpect, 42};
+        const Expected const_with_error{unexpect, 1337};
+        assert(with_error.error_or(convertible{1}) == 42 + 2);
+        assert(const_with_error.error_or(convertible{1}) == 1337 + 2);
+        static_assert(noexcept(with_error.error_or(convertible{1})) == should_be_noexcept);
+        static_assert(noexcept(const_with_error.error_or(convertible{1})) == should_be_noexcept);
+
+        assert(move(with_error).error_or(convertible{1}) == 42 + 3);
+        assert(move(const_with_error).error_or(convertible{1}) == 1337 + 2);
+        static_assert(noexcept(move(with_error).error_or(convertible{1})) == should_be_noexcept);
+        static_assert(noexcept(move(const_with_error).error_or(convertible{1})) == should_be_noexcept);
+
+        const convertible input{2};
+        Expected with_value{in_place, 42};
+        const Expected const_with_value{in_place, 1337};
+        assert(with_value.error_or(convertible{1}) == 1 + 5);
+        assert(const_with_value.error_or(input) == 2 + 4);
+        static_assert(noexcept(with_value.error_or(convertible{1})) == should_be_noexcept);
+        static_assert(noexcept(const_with_value.error_or(input)) == should_be_noexcept);
+
+        assert(move(with_value).error_or(convertible{1}) == 1 + 5);
+        assert(move(const_with_value).error_or(input) == 2 + 4);
+        static_assert(noexcept(move(with_value).error_or(convertible{1})) == should_be_noexcept);
+        static_assert(noexcept(move(const_with_value).error_or(input)) == should_be_noexcept);
+    }
+}
+
+constexpr void test_error_or() noexcept {
+    test_error_or<IsNothrowConstructible::Not, IsNothrowConvertible::Not>();
+    test_error_or<IsNothrowConstructible::Not, IsNothrowConvertible::Yes>();
+    test_error_or<IsNothrowConstructible::Yes, IsNothrowConvertible::Not>();
+    test_error_or<IsNothrowConstructible::Yes, IsNothrowConvertible::Yes>();
+}
+
+constexpr void test_monadic() {
+    {
+        expected<Thingy, int> engaged{Thingy{11}};
+        expected<Thingy, int> unengaged{std::unexpect, 22};

```

Comment:
Any reason to qualify? There's `using namespace` on file scope. Also occurs below.